Feature/android/thirdpartypositioning (#694)
MapsPeople/docs: src/android/v3/guides/third-party-positioning/google-fused-location-integration.md

---
title: Use Google Fused Location Provider with MapsIndoors
eleventyNavigation:
  parent: android-v3-guides-use-third-party-positioning-with-MapsIndoors
  title: Use Google Fused Location Provider with MapsIndoors
  key: android-v3-use-Google-Fused-Location-Provider-with-MapsIndoors
  order: 230
---

> For Google Fused Location Provider you will need to create a positioning implementation to have the positions received from the api communicate with the MapsIndoors SDK.

Reviewer: "api" should be capitalized

> For Google Fused Location Provider you will need to create a positioning implementation to have the positions received from the API communicate with the MapsIndoors SDK.

Reviewer 2: ```suggestion
For Google Fused Location Provider, you need to create a positioning implementation which enables communicating the positions received from the API with the MapsIndoors SDK.
```

        });
    }
}

void onGooglePositioningPermissionsGiven() {
    MapsIndoors.setPositionProvider(mGooglePositioningProvider);
    MapsIndoors.startPositioning();
    mMapControl.showUserPosition(true);

    mGooglePositioningProvider.addOnPositionUpdateListener(new OnPositionUpdateListener() {
        @Override
        public void onPositioningStarted(@NonNull @NotNull PositionProvider positionProvider) {

        }

        @Override
        public void onPositionFailed(@NonNull @NotNull PositionProvider positionProvider) {

        }

        @Override
        public void onPositionUpdate(@NonNull @NotNull PositionResult positionResult) {
            mActivity.runOnUiThread(() -> {
                mMapControl.getPositionIndicator().setIconFromDisplayRule( new LocationDisplayRule.Builder( "BlueDotRule" )
                        .setVectorDrawableIcon(android.R.drawable.presence_invisible, 23, 23 )
                        .setTint(Color.BLUE)
                        .setShowLabel(true)
                        .setLabel("You")
                        .setLabel(null)
                        .build());
            });
        }
    });
}
```

</mi-tab-panel>
</mi-tabs>

> All that is left to do now is to start this up after initialising our mapControl.

Reviewer: Initializing, in proper US English :P
